dali/pipeline/executor/executor.cc

```diff
@@ -644,7 +649,45 @@ bool Executor<WorkspacePolicy, QueuePolicy>::HasConditionals() const {
   return has_conditionals_;
 }
 
+template<typename WorkspacePolicy, typename QueuePolicy>
+void Executor<WorkspacePolicy, QueuePolicy>::InitCheckpointing() {
+  if (!checkpointing_)
+    return;
+
```

Comment:
Presumingly, we also need a check for the execution policy (we won't support the separated queues).